File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/ncp/Boost.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.ncp;

import net.ccbluex.liquidbounce.event.MoveEvent;
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode;
import net.ccbluex.liquidbounce.utils.MovementUtils;
import net.minecraft.util.AxisAlignedBB;

public class Boost extends SpeedMode {

    public Boost() {
        super("Boost");
    }

    private int motionDelay;
    private float ground;

    @Override
    public void onMotion() {
        double speed = 3.1981D;
        double offset = 4.69D;
        boolean shouldOffset = true;

        for(final Object o : mc.theWorld.getCollidingBoundingBoxes(mc.thePlayer, mc.thePlayer.getEntityBoundingBox().offset(mc.thePlayer.motionX / offset, 0.0D, mc.thePlayer.motionZ / offset))) {
            if(o instanceof AxisAlignedBB) {
                shouldOffset = false;
                break;
            }
        }

        if(mc.thePlayer.onGround && ground < 1.0F)
            ground += 0.2F;

        if(!mc.thePlayer.onGround)
            ground = 0.0F;

        if(ground == 1.0F && shouldSpeedUp()) {
            if(!mc.thePlayer.isSprinting())
                offset += 0.8D;

            if(mc.thePlayer.moveStrafing != 0F) {
                speed -= 0.1D;
                offset += 0.5D;
            }

            if(mc.thePlayer.isInWater())
                speed -= 0.1D;

            motionDelay += 1;

            switch(motionDelay) {
                case 1:
                    mc.thePlayer.motionX *= speed;
                    mc.thePlayer.motionZ *= speed;
                    break;
                case 2:
                    mc.thePlayer.motionX /= 1.458D;
                    mc.thePlayer.motionZ /= 1.458D;
                    break;
                case 4:
                    if(shouldOffset)
                        mc.thePlayer.setPosition(mc.thePlayer.posX + mc.thePlayer.motionX / offset, mc.thePlayer.posY, mc.thePlayer.posZ + mc.thePlayer.motionZ / offset);

                    motionDelay = 0;
                    break;
            }
        }
    }

    @Override
    public void onUpdate() {
    }

    @Override
    public void onMove(MoveEvent event) {
    }

    private boolean shouldSpeedUp() {
        return !mc.thePlayer.isInWater() && (!mc.thePlayer.isOnLadder()) && !mc.thePlayer.isSneaking() && MovementUtils.isMoving();
    }
}

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speedmodes/ncp/Boost.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speedmodes.ncp

import net.ccbluex.liquidbounce.features.module.modules.movement.speedmodes.SpeedMode
import net.ccbluex.liquidbounce.utils.MovementUtils.isMoving

object Boost : SpeedMode("Boost") {
    private var motionDelay = 0
    private var ground = 0f
    override fun onMotion() {
        val player = mc.thePlayer ?: return

        var speed = 3.1981
        var offset = 4.69
        var shouldOffset = true

        if (mc.theWorld.getCollidingBoundingBoxes(player, player.entityBoundingBox.offset(player.motionX / offset, 0.0, player.motionZ / offset)).isNotEmpty()) {
            shouldOffset = false
        }

        if (player.onGround && ground < 1f)
            ground += 0.2f
        if (!player.onGround)
            ground = 0f

        if (ground == 1f && shouldSpeedUp()) {
            if (!player.isSprinting)
                offset += 0.8

            if (player.moveStrafing != 0f) {
                speed -= 0.1
                offset += 0.5
            }
            if (player.isInWater)
                speed -= 0.1


            motionDelay += 1
            when (motionDelay) {
                1 -> {
                    player.motionX *= speed
                    player.motionZ *= speed
                }
                2 -> {
                    player.motionX /= 1.458
                    player.motionZ /= 1.458
                }
                4 -> {
                    if (shouldOffset) player.setPosition(player.posX + player.motionX / offset, player.posY, player.posZ + player.motionZ / offset)
                    motionDelay = 0
                }
            }
        }
    }


    private fun shouldSpeedUp() =
        !mc.thePlayer.isInLava && !mc.thePlayer.isOnLadder && !mc.thePlayer.isSneaking && isMoving
}